### PR TITLE
Snowflake show statement bug fixes

### PR DESCRIPTION
**Pull request by mhaley-miovision**
Related to https://github.com/sqlfluff/sqlfluff/pull/686.
I jumped the gun a bit and missed your comment here https://github.com/sqlfluff/sqlfluff/pull/686#discussion_r558995519.

I went ahead and added some additional tests which uncovered some bugs.  Sorry @NiallRees, but the fixes are here.

### CODE REVIEW

File: src/sqlfluff/core/dialects/dialect_snowflake.py [hunk 1]
```diff
@@ -561,7 +563,7 @@ class ShowStatementSegment(BaseSegment):
         OneOf("TERSE", optional=True),
         _object_types_plural,
         OneOf("HISTORY", optional=True),
-        Sequence("LIKE", Ref("ObjectReferenceSegment"), optional=True),
+        Sequence("LIKE", Ref("Expression_A_Grammar"), optional=True),
```
Comment (NiallRees) [inline src/sqlfluff/core/dialects/dialect_snowflake.py:566]: How come Expression_A_Grammar not ObjectReferenceSegment?

Comment (mhaley-miovision) [reply src/sqlfluff/core/dialects/dialect_snowflake.py:566]: It is really a string expression. Possible values could be `LIKE '%MYTABLE%'`. I guess we could use `QuotedLiteralSegment`, which does pass the tests. What do you think?

Comment (NiallRees) [reply src/sqlfluff/core/dialects/dialect_snowflake.py:566]: I think that's a better suggestion. Expression_A_Grammar is a very flexible grammar which doesn't need to be used here so I'd keep it simple with `QuotedLiteralSegment`.

Comment (NiallRees) [inline src/sqlfluff/core/dialects/dialect_snowflake.py:566]: Here too @mhaley-miovision

**Review by NiallRees — CHANGES_REQUESTED** [review]

**Review by NiallRees — APPROVED** [review]

**Pull request merged**